File: src/types/payment.ts
export interface Payment {
  id: number;
  member_id?: number | null;
  member_name?: string | null;
  amount: number;
  payment_method: string;
  payment_type?: string | null;
  status: string;
  reference?: string | null;
  notes?: string | null;
  payment_date: string;
  created_by?: number | null;
  created_at: string;
  is_deleted: boolean;
}

export interface CreatePaymentInput {
  member_id?: number | null;
  amount: number;
  payment_method: string;
  payment_type?: string | null;
  status?: string | null;
  reference?: string | null;
  notes?: string | null;
  payment_date?: string | null;
}

export interface MemberBalance {
  member_id: number;
  member_name: string;
  phone?: string | null;
  total_credit: number;
  total_paid: number;
  balance: number;
  payment_count: number;
}

export const PAYMENT_METHODS = ["cash", "bank_transfer", "card", "other"] as const;
export const PAYMENT_STATUSES = ["completed", "pending", "cancelled", "refunded"] as const;

File: src-tauri/src/models/payment.rs
use serde::{Deserialize, Serialize};

#[derive(Serialize, Deserialize, Debug, Clone)]
pub struct Payment {
    pub id: i64,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub member_id: Option<i64>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub member_name: Option<String>,
    pub amount: f64,
    pub payment_method: String,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub payment_type: Option<String>,
    pub status: String,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub reference: Option<String>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub notes: Option<String>,
    pub payment_date: String,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub created_by: Option<i64>,
    pub created_at: String,
    pub is_deleted: bool,
}

#[derive(Serialize, Deserialize, Debug, Clone, Default)]
pub struct CreatePaymentInput {
    pub member_id: Option<i64>,
    pub amount: f64,
    pub payment_method: String,
    pub payment_type: Option<String>,
    pub status: Option<String>,
    pub reference: Option<String>,
    pub notes: Option<String>,
    pub payment_date: Option<String>,
}

#[derive(Serialize, Deserialize, Debug, Clone)]
pub struct MemberBalance {
    pub member_id: i64,
    pub member_name: String,
    pub phone: Option<String>,
    /// Amount owed at sale time (sum of unpaid credit portions) — Σ(sales.total - sales.paid).
    pub total_credit: f64,
    /// Sum of payments received (completed) against the member.
    pub total_paid: f64,
    /// Outstanding balance = total_credit - total_paid.
    pub balance: f64,
    pub payment_count: i64,
}
